plugindocs.rb

```diff
@@ -1,24 +1,32 @@
 require "clamp"
-require "json"
 require "fileutils"
+require "json"
+require "pmap" # Enumerable#peach
+require "set"
+require "stud/try"
 require "time"
+require "thread" # Mutex
 require "yaml"
-require "stud/try"
-require "pmap" # Enumerable#peach
 
 require_relative 'lib/logstash-docket'
 
 class PluginDocs < Clamp::Command
   option "--output-path", "OUTPUT", "Path to the top-level of the logstash-docs path to write the output.", required: true
   option "--main", :flag, "Fetch the plugin's docs from main instead of the version found in PLUGINS_JSON", :default => false
   option "--settings", "SETTINGS_YAML", "Path to the settings file.", :default => File.join(File.dirname(__FILE__), "settings.yml"), :attribute_name => :settings_path
-  option("--parallelism", "NUMBER", "for performance", default: 4) { |v| Integer(v) }
+  option("--parallelism", "NUMBER", "for performance", default: 1) { |v| Integer(v) }
```

Comment:
Since we constrain where needed in the non-versioned doc gen, we don't need to change the default. Having a default of 4 makes the versioned-plugin-reference _much_ faster.

```suggestion
  option("--parallelism", "NUMBER", "for performance", default: 4) { |v| Integer(v) }
```